inst/testfiles/binseg_interface/binseg_interface_DeepState_TestHarness.cpp

```diff
@@ -0,0 +1,53 @@
+#include <fstream>
+#include <RInside.h>
+#include <iostream>
+#include <RcppDeepState.h>
+#include <qs.h>
+#include <DeepState.hpp>
+
+RInside Rinstance;
+
+/** FUNCTION SIGNATURE */
+Rcpp::List binseg_interface(const Rcpp::NumericVector data_vec, const Rcpp::NumericVector weight_vec, 
+const int max_segments, const int min_segment_length, const std::string distribution_str, 
+const std::string container_str, const Rcpp::LogicalVector is_validation_vec, const Rcpp::NumericVector position_vec);
+
+Rcpp::LogicalVector RcppDeepState_LogicalVector(int rand_size){
+  Rcpp::LogicalVector rand_vec(rand_size);
+    for(int i = 0 ; i < rand_size;i++){      
+      rand_vec[i] = DeepState_IntInRange(0,1);  
+    }
+
+  return rand_vec;
+}
+
+#define INPUTS \
```

Comment:
you could use a macro like this to avoid duplication in definition of data /generator code.

Reply:
@FabrizioSandri

Reply:
@tdhock I have revised the harness creation procedure of RcppDeepState and made some additional improvements, including the use of macros. You can find these new changes in https://github.com/FabrizioSandri/RcppDeepState/pull/16. 

Before testing this change I have to merge https://github.com/FabrizioSandri/RcppDeepState/pull/16 into the main branch. I will wait for your code review before merging these modifications, so that if something is wrong I can simply adjust.